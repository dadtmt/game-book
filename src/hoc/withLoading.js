import React from 'react'
import { branch, renderComponent } from 'recompose'

const Loader = () => <div>Loading...</div>

const withLoading = branch(
  ({data}) => data.loading,
  renderComponent(Loader)
)

export default withLoading
