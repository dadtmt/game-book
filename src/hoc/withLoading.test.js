import React from 'react'
import renderer from 'react-test-renderer';

import withLoading from './withLoading'


const Loaded = ({data}) => <div>{data.loaded}</div>

const ComponentWithLoading = withLoading(Loaded)

it('renders loading', () => {
  const data = {
    loading: true
  }
  const tree = renderer
    .create(<ComponentWithLoading data={data} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders loaded', () => {
  const data = {
    loaded: 'component loaded'
  }
  const tree = renderer
    .create(<ComponentWithLoading data={data} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
