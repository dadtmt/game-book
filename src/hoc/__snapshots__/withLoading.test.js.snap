// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loaded 1`] = `
<div>
  component loaded
</div>
`;

exports[`renders loading 1`] = `
<div>
  Loading...
</div>
`;
